backends/p4tools/common/core/target.cpp

```diff
@@ -76,12 +77,13 @@ const IR::Expression *Target::createTargetUninitialized(const IR::Type *type,
     return IR::getDefaultValue(type);
 }
 
-Target::Target(std::string toolName, std::string deviceName, std::string archName)
+Target::Target(std::string_view toolName, const std::string &deviceName,
+               const std::string &archName)
     : toolName(toolName), spec(deviceName, archName) {
     // Register this instance.
     BUG_CHECK(!registry[spec].count(toolName), "Already registered %1%/%2% instance for %3%",
               deviceName, archName, toolName);
-    registry[spec][toolName] = this;
+    registry[spec][toolName.data()] = this;
```

Comment:
```suggestion
    registry[spec][toolName] = this;

Reply:
Lookups do not seem to work for transparent comparators after all.

Reply:
Oh, I accidentally looked at the first level map in `registry` where the key is `Target::Spec`. Then this is weird, it should work if I understand C++ documentation correctly :-/.